fs/zbd_zenfs.h

```diff
@@ -90,32 +90,46 @@ class Zone {
   void CloseWR(); /* Done writing */
 };
 
+// Abstract class as interface.
+// operator() must be overrided in order to execute the function.
 class BackgroundJob {
  public:
-  BackgroundJob(std::function<int(void *)> fn, void *arg)
-       : fn_(fn), arg_(arg) {}
-   std::function<int(void *)> fn_;
-  void *arg_;
-  virtual void operator()() { fn_(arg_); }
+  virtual void operator()() = 0;
   virtual ~BackgroundJob() {}
 };
 
-class ErrorHandlingBGJob : public BackgroundJob {
+// For simple jobs that needs no return value nor argument.
+class SimpleJob : public BackgroundJob {
+ public:
+  std::function<void()> fn_;
+  // No default allowed.
+  SimpleJob() = delete;
+  SimpleJob(std::function<void()> fn) : fn_(fn) {}
+  virtual void operator()() override { fn_(); }
+  virtual ~SimpleJob() {}
+};
+
+template<typename arg_t, typename ret_t>
```

Comment:
We use capital letters for template args in google code style IIRC.

Reply:
LOL, ok, ok.